posts/2021-04-16-mp-context-propagration-21004.adoc

```diff
@@ -0,0 +1,387 @@
+* link:{url-prefix}/docs/latest/reference/config/authFilter.html[Open Liberty Documentation for Authentication Filter (AuthFilter)]
+
+[#requestTiming]
+== Control thread dump collection for hung requests in the Request Timing feature
+
+The Request Timing feature (`requestTiming-1.0`) provides diagnostic information when the duration of any request exceeds the configured threshold. It provides a way to monitor requests with respect to time. The feature can automatically detect slow and hung requests and provide detailed diagnostic information; warning messages, thread stacks, and the creation of thread dumps.
+
+When a hung request is detected in the Request Timing feature, a warning message is written in the messages log file along with a dump of the events that happened during the request. Following that, a set of three thread dumps will be initiated, 1 minute apart. After the completion of the three thread dumps, further set of three thread dumps are created only if new requests are detected to be hanging.
+
+Some operations teams do not want so many thread dumps to be generated due to performance overhead on requests that are known to be long. In previous Open Liberty releases, there was no option to disable the thread dumps from being generated.
+
+In 21.0.0.4, you can now control whether the Request Timing feature collects thread dumps. By setting the new `enableThreadDumps` Request Timing server configuration attribute to false, thread dumps will not be created during hung requests. If `enableThreadDumps` is set to true or not specified at all, thread dumps will still be created.
+   
+The new Request Timing server configuration attribute can be configured in your server.xml as follows:
+
+[source, xml]
+----
+<requestTiming includeContextInfo="true" slowRequestThreshold="120s" hungRequestThreshold="10s" sampleRate="1" enableThreadDumps="false"></requestTiming>`
+----
+
+
+The `enableThreadDumps` server configuration attribute can also be used in embedded Request Timing sub-elements: 
+`<servletTiming/>` or `<jdbcTiming/>`, as follows:
+
+[source, xml]
+----
+<requestTiming includeContextInfo="true" slowRequestThreshold="120s" hungRequestThreshold="10s" sampleRate="1">
+    <servletTiming appName="MyApp" servletName="MyServletApp" slowRequestThreshold="100s" hungRequestThreshold="5s" enableThreadDumps="false"/>
+</requestTiming>`
+----
+
+NOTE: An embedded `<servletTiming/>` or `<jdbcTiming/>` configuration in the server.xml file overrides the configured slow and hung request threshold that are defined in `<requestTiming/>`.  
+
+For more information on the Request Timing feature, please refer to the following documentations:
+- link:{url-prefix}/docs/latest/reference/feature/requestTiming-1.0.html[Open Liberty Documentation on requestTiming-1.0 feature]
+- link:{url-prefix}/docs/latest/reference/config/requestTiming.html[Open Liberty Documentation on requestTiming Configuration]
```

Comment:
Not rendering properly on draft site (single line instead of the links on their own lines):
`For more information on the Request Timing feature, please refer to the following documentations: - Open Liberty Documentation on requestTiming-1.0 feature - Open Liberty Documentation on requestTiming Configuration`

Reply:
Done